simulacrum/types.py

```diff
@@ -0,0 +1,125 @@
+"""Functions for generating random data."""
+
+from uuid import uuid4
+import logging
+import sys
+import datetime
+import numpy as np
+import pandas as pd
+from faker import Faker
+
+FAKE = Faker()
+DT_CLASSES = (datetime.datetime, datetime.date)
+
+def name_data(length):
+    """Faker names series"""
+    return pd.Series([FAKE.name() for _ in range(length)])
+
+def text_data(length, max_nb_chars=200):
+    """Faker text series"""
+    return pd.Series([FAKE.text(max_nb_chars) for _ in range(length)])
+
+def address_data(length):
+    """Faker address series"""
+    return pd.Series([FAKE.address() for _ in range(length)])
+
+def num_data(length, minimum=0, maximum=1):
```

Comment:
Hum it's a breaking change. Indeed @jbrambleDC originally used min and max as you said in your pull request description...
I don't really care about these parameter names. @jbrambleDC have to choose.

However, A workaround could be to add a high order function (typically a decorator) on num_data and num_int to ensure a compatibility mode on parameters (and print a warning to indicate a deprecation of min and max).
Example:

````python
def ensure_num_compatibility(func):
    """Make sure num_data or num_int will be compatible with minimum and maximum parameters."""
    def compatibility_func(*args, **kwargs):
        compatible_kwargs = dict(kwargs)
        def replace_in_kwargs(parameter_to_replace, new_parameter):
            if parameter_to_replace in kwargs:
                compatible_kwargs[new_parameter] = kwargs[parameter_to_replace]
                del compatible_kwargs[parameter_to_replace]
        replace_in_kwargs('min', 'minimum')
        replace_in_kwargs('max', 'maximum')
        return func(*args, **compatible_kwargs)
    return compatibility_func


@ensure_num_compatibility
def num_int(length, minimum=0, maximum=100):
    """Random integers"""
    return pd.Series(np.random.random_integers(minimum, maximum, length))


num_int(10, min=10, max=22)

````